ext/xml/xml.c

```diff
@@ -1102,11 +1080,59 @@ static bool php_xml_check_string_method_arg(
 	parser_handler_fcc->function_handler = method_ptr;
 	parser_handler_fcc->calling_scope = ce;
 	parser_handler_fcc->called_scope = ce;
-	parser_handler_fcc->object = parser->object;
+	parser_handler_fcc->object = object;
 
 	return true;
 }
 
+#define PHP_XML_CHECK_NEW_THIS_METHODS(parser_to_check, new_this_obj, fcc_field) \
+	if (ZEND_FCC_INITIALIZED(parser_to_check->fcc_field) && parser_to_check->fcc_field.object == parser_to_check->object) { \
+		zend_string *method_name = zend_string_copy(parser_to_check->fcc_field.function_handler->common.function_name); \
```

Comment:
Doesn't this leak with non-interned strings? (testable via: `str_repeat("methodname", mt_rand(1,1))` note the mt_rand to defeat SCCP optimization)

Reply:
I wouldn't think so? But I'll add a test

Reply:
No test needed, could repro with current tests. Fix committed.